sp-validation/creator-node/healthChecks.js

```diff
@@ -0,0 +1,153 @@
+const axios = require('axios')
+const Web3 = require('web3')
+const web3 = new Web3()
+const { promisify } = require('util')
+
+const crypto = require('crypto')
+const assert = require('assert')
+
+const PRIVATE_KEY = process.env.delegatePrivateKey
+const CREATOR_NODE_ENDPOINT = process.env.creatorNodeEndpoint
+
+/**
+ * Parses the environment variables and command line args
+ * export creatorNodeEndpoint=http://creatornode.domain.com
+ * export delegatePrivateKey=f0b743ce8adb7938f1212f188347a63...
+ * NOTE: DO NOT PREFIX PRIVATE KEY WITH 0x
+ */
+function parseEnvVarsAndArgs () {
+  if (!CREATOR_NODE_ENDPOINT || !PRIVATE_KEY) {
+    let errorMsg = `creatorNodeEndpoint [${CREATOR_NODE_ENDPOINT}] or delegatePrivateKey [${PRIVATE_KEY}] have not been exported. `
+    errorMsg += "Please export environment variables 'delegatePrivateKey' (without leading 0x) and 'creatorNodeEndpoint' with https."
+    throw new Error(errorMsg)
+  }
+}
+
+async function healthCheck () {
+  let requestConfig = {
+    url: `${CREATOR_NODE_ENDPOINT}/health_check`,
+    method: 'get',
+    responseType: 'json'
+  }
+  let resp = await axios(requestConfig)
+  let data = resp.data
+  assert.deepStrictEqual(resp.status, 200)
+  assert.deepStrictEqual(data.data.healthy, true)
+  assert.ok(
+    data.data.selectedDiscoveryProvider !== null && data.data.selectedDiscoveryProvider !== undefined,
+    `Selected discovery provider should not be null or undefined`
+  )
+  assert.ok(
+    data.data.signature !== null && data.data.signature !== undefined,
+    `Signature should not be null or undefined`
+  )
+  console.log('✓ Health check passed successfully')
+}
+
+async function healthCheckDB () {
+  let requestConfig = {
+    url: `${CREATOR_NODE_ENDPOINT}/db_check`,
+    method: 'get',
+    responseType: 'json'
+  }
+  let resp = await axios(requestConfig)
+  assert.deepStrictEqual(resp.status, 200)
+  console.log('✓ DB health check passed successfully')
+}
+
+async function healthCheckDisk () {
+  let requestConfig = {
+    url: `${CREATOR_NODE_ENDPOINT}/disk_check`,
+    method: 'get',
+    responseType: 'json'
+  }
+  let resp = await axios(requestConfig)
+  let data = resp.data
+  console.log(data)
+  assert.deepStrictEqual(resp.status, 200)
+  const [size, magnitude] = data.data.available.split(' ')
+  assert.deepStrictEqual(magnitude, 'TB')
+  assert.ok(parseFloat(size) > 1.5, 'Minimum available disk space should be 1.5 TB')
+  console.log('✓ Disk health check passed successfully')
```

Comment:
i like this checkmark idea but next time instead of manually adding it everywhere it would be nicer to just prefix and have a script-level logging standard. non blocking